### ISSUE TITLE
Method to parse LitStr into syntax tree node

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/syn#342]
Maybe something like:

```rust
impl LitStr {
    fn parse<T: Synom>(&self) -> Result<T, ParseError>;
}
```

This is common when parsing [`Meta`](https://docs.rs/syn/0.12/syn/enum.Meta.html) items in a proc macro. Importantly, the `parse` method would need to respan all the tokens in the output to have the same span as the LitStr input so that errors point to the right place.

[Possible implementation from serde_derive.](https://github.com/serde-rs/serde/blob/1bdf5ecec40ae9d4b77b321341a4f5c9cd4cb6ba/serde_derive_internals/src/attr.rs#L1355-L1382)

**Comment by mystor** [comment]
Yeah, this seems fine to me - we implement it by parsing the string into a `TokenStream`, respanning that `TokenStream`, and then parsing it?

**Comment by dtolnay** [comment]
Yes I think that would be best.

**Comment by dtolnay** [comment]
Implemented in d53ac2b150046116cc7971d7055319feda7418c0.
